test/mpi/spawn/join.c

```diff
@@ -88,6 +88,8 @@ int main(int argc, char *argv[])
         MPI_Send(hostname, namelen + 1, MPI_CHAR, 0, 0, MPI_COMM_WORLD);
         MPI_Send(&port, 1, MPI_INT, 0, 1, MPI_COMM_WORLD);
 
+        MPI_Barrier(MPI_COMM_WORLD);
```

Comment:
I'm not sure how an implemention could do anything differently. MPI_SEND would have to guarantee that the messages have been put on the wire in order for this to work. The standard only guarantees that the buffers are safe to reuse and may be finished later (maybe at finalize). 

The receiver can't do anything to help either.